dask/dataframe/tests/test_groupby.py

```diff
@@ -430,6 +430,14 @@ def test_series_groupby_propagates_names():
     assert_eq(result, expected)
 
 
+def test_series_groupby_with_named_index():
```

Comment:
Given that there are different code paths for single-partition and multi-partition dataframes, perhaps we should parameterize over `npartitions`? Could also parameterize over `cumprod` and `cumcount` while we are at it